### PR TITLE
Have an option to submit diagnostics on any failure

### PR DESCRIPTION
**Pull request by jvimal-eg**

### CODE REVIEW

File: cleanup.js [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
 const { shell } = require('./shell.js');
 
+if (core.getState("EG_FAILED") == "true") {
+    if (core.getInput('submit_diagnostics_on_failure') == "true") {
+        try {
+            shell('egctl advanced log-upload');
+        } catch (error) { }
+    }
+    try {
+        shell('cat /var/log/edgeguardian/datapath.log || true;')
```
Comment (vadim-eg) [inline cleanup.js:11]: This may be considered too revealing, we may need to do some sanitation.

**Review by vadim-eg — APPROVED** [review]
lgtm

File: main.js [hunk 2]
```diff
@@ -25,5 +25,6 @@ async function main() {
 try {
     main();
 } catch (error) {
+    core.saveState("EG_FAILED", "true");
```
Comment (theron-eg) [inline main.js:28]: Does saveState only take a string?  I'm curious whether you can pass `true` as opposed to `"true"`.

**Review by theron-eg — APPROVED** [review]
lgtm!

File: main.js (hunk 2 above)
Comment (jvimal-eg) [reply main.js:28]: Looking at https://github.com/actions/toolkit/blob/master/packages/core/src/core.ts#L208, I see it being passed around as strings.

File: cleanup.js (hunk 1 above)
Comment (jvimal-eg) [reply cleanup.js:11]: Good point, I am erring on the side of debugging now.  I can relax this once we debug the current issue.

File: main.js (hunk 2 above)
Comment (theron-eg) [reply main.js:28]: Well, good news is JS lets you use them interchangeably most of the time!

**Pull request merged**